go/mcap/writer.go

```diff
@@ -445,25 +444,45 @@ func (w *Writer) flushActiveChunk() error {
 	// when writing a chunk, we don't go through writerecord to avoid needing to
 	// materialize the compressed data again. Instead, write the leading bytes
 	// then copy from the compressed data buffer.
-	recordlen := 1 + 8 + msglen
-	if len(w.chunk) < recordlen {
-		w.chunk = make([]byte, recordlen*2)
-	}
-	offset, err := putByte(w.chunk, byte(OpChunk))
+	_, err = w.w.Write([]byte{byte(OpChunk)})
```

Comment:
I would expect this could change the dynamics depending on whether the user is passing a buffered or unbuffered writer. The previous implementation effectively buffered writes no matter what kind of writer got passed in, while this one seems like it will do more syscalls if the writer passed is unbuffered. I don't think that explains the lack of performance changes though since the benchmark is writing to ram.